lib/rules/selector-not-notation/index.mjs

```diff
@@ -82,43 +90,49 @@ const rule = (primary, _, context) => {
 			selectorRoot.walkPseudos((pseudo) => {
 				if (!isNot(pseudo)) return;
 
+				const source = getSource(pseudo, ruleNode);
+
 				if (primary === 'complex') {
 					const prev = pseudo.prev();
 					const hasConsecutiveNot = prev && isNot(prev);
 
 					if (!hasConsecutiveNot) return;
 
-					if (context.fix) return fixComplex(prev);
+					if (context.fix) {
+						return addFixer(result, ruleName, {
+							source,
+							callback: fixer,
+							args: { cb: fixComplex, nodes: { root: selectorRoot, rule: ruleNode, pseudo: prev } },
+						});
+					}
 				} else {
 					const selectors = pseudo.nodes;
 
 					if (isSimple(selectors)) return;
 
-					const mustFix =
-						context.fix &&
-						selectors.length > 1 &&
-						selectors[1] &&
-						(selectors[1].nodes.length === 0 || selectors.every(({ nodes }) => nodes.length === 1));
-
-					if (mustFix) return fixSimple(pseudo);
+					const second = selectors.length > 1 && selectors[1];
+					const isFixable =
+						second && (!second.nodes.length || selectors.every(({ nodes }) => nodes.length === 1));
+
+					if (context.fix) {
+						return addFixer(result, ruleName, {
+							source,
+							callback: fixer,
+							args: { cb: fixSimple, nodes: { root: selectorRoot, rule: ruleNode, pseudo } },
+							unfixable: !isFixable,
```

Comment:
note

We don't bail out on the fixer addition because we will eventually have to expose the fix data regardless of the fixability status.
i.e. it's simpler to do it that way than having different paths down the line